lib/migrations/migrator.js

```diff
@@ -138,11 +138,19 @@ class Migrator {
 
   async migrate() {
     const version = await this.ldb.get(this.layout.V.encode());
-
-    if (version === null && this.migrateFlag !== -1)
-      throw new Error('Database does not exist.');
+    const lastID = this.getLastMigrationID();
 
     if (version === null) {
+      if (this.migrateFlag !== -1) {
+        if (this.migrateFlag !== lastID) {
+          throw new Error(
+            `Migrate flag ${this.migrateFlag} does not match last ID: ${lastID}`
+          );
+        } else {
```

Comment:
you can get rid of else here, above will throw anyway

Reply:
right thanks!